runtime/runtime/src/prefetch.rs

```diff
@@ -157,23 +164,31 @@ impl TriePrefetcher {
         self.prefetch_api.clear_data();
     }
 
-    fn prefetch_trie_key(&self, trie_key: TrieKey) -> Result<(), ()> {
-        let queue_full = self.prefetch_api.prefetch_trie_key(self.trie_root, trie_key).is_err();
-        if queue_full {
-            self.prefetch_queue_full.inc();
-            debug!(target: "prefetcher", "I/O scheduler input queue full, dropping prefetch request");
-            Err(())
-        } else {
-            self.prefetch_enqueued.inc();
-            Ok(())
-        }
+    fn prefetch_trie_key(&self, trie_key: TrieKey) -> Result<(), PrefetchError> {
+        let res = self.prefetch_api.prefetch_trie_key(self.trie_root, trie_key);
+        match res {
+            Err(PrefetchError::QueueFull) => {
+                self.prefetch_queue_full.inc();
+                debug!(target: "prefetcher", "I/O scheduler input queue is full, dropping prefetch request");
+            }
+            Err(PrefetchError::QueueDisconnected) => {
+                // This shouldn't have happened, hence logging warning here
+                warn!(target: "prefetcher", "I/O scheduler input queue is disconnected, dropping prefetch request");
+            }
+            Ok(_) => self.prefetch_enqueued.inc(),
```

Comment:
This makes sure that if the type changes, the compiler will tell us all the places we need to handle the situation.

```suggestion
            Ok(()) => self.prefetch_enqueued.inc(),
```